subscription-engine/kafka-ws/app/WSServer.js

```diff
@@ -0,0 +1,303 @@
+const { Logger } = require('@dojot/microservice-sdk');
+const WebSocket = require('ws');
+const url = require('url');
+const { v4: uuidv4 } = require('uuid');
+
+const { WSError } = require('./Errors').Errors;
+const { ErrorCodes } = require('./Errors');
+const { ProcessingRuleManager } = require('./ProcessingRule');
+const { WhereParser } = require('./WhereProcessing');
+
+const RedisExpirationMgmt = require('./Redis/RedisExpireMgmt');
+
+const KafkaTopicsCallbacksMgmt = require('./Kafka/KafkaTopicsConsumerCallbacksMgmt');
+const {
+  checkAndParseURLPathname,
+  parseTenantAndExpTimeFromToken,
+  checkTopicBelongsTenant,
+  isObjectEmpty,
+  addTimeFromNow,
+} = require('./Utils');
+const { server: serverConfig } = require('./Config');
+
+const logger = new Logger();
+
+
+/**
+ * Gets the kafka topic from the given url
+ * TODO: put this function in a more appropriate place, as a class to handle connections
+ *
+ * @param {string} fullUrl
+ * @param {funcion} cbError callback if error occurs
+ */
+const getKafkaTopicFromPathname = (fullUrl, cbError) => {
+  let parsedRoute = null;
+  // check if the url pathname has the expected pattern with the topic
+  try {
+    parsedRoute = checkAndParseURLPathname(fullUrl, '/v1/websocket/:topic');
```

Comment:
`'/v1/websocket/:topic'`
In an API versioning by URL it is interesting to have a structure break, to separate what belongs to `v1` and what would belong to a possible `v2`. but for now there is no need.

Reply:
Yes, It's a good point